### ISSUE TITLE
Pool vectors for photons to improve performance

### ISSUE THREAD

**Issue by jbphet** [phetsims/greenhouse-effect#304]
While working on phet-io state,  @zepumph suggested that photon movement and state setting might be more efficient if we use pooling for the vectors. 

There are apparently some good examples in Projectile Motion, search for `Vector2.pool.create(` to see them.

**Comment by zepumph** [comment]
Yes this seems like it would be pretty dang easy to implement inside of Photon.ts, and would give us some gains. Over in #301 we didn't see a bunch of performance problems when setting a couple thousand photons to the state (<500ms on average), but if this is simple, since we KNOW just how many objects are being used and discarded, it seems worth while. I would have it on a lowish priority, and its potential simplicity would up it a medium priority.
